memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateConversationRequest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.memory.action.conversation;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+import static org.opensearch.ml.common.conversation.ConversationalIndexConstants.APPLICATION_TYPE_FIELD;
+import static org.opensearch.ml.common.conversation.ConversationalIndexConstants.META_NAME_FIELD;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.core.common.io.stream.InputStreamStreamInput;
+import org.opensearch.core.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentParser;
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+public class UpdateConversationRequest extends ActionRequest {
+    private String conversationId;
+    private Map<String, Object> updateContent;
+
+    private static final Set<String> allowedList = new HashSet<>(Arrays.asList(META_NAME_FIELD, APPLICATION_TYPE_FIELD));
```

Comment:
Do we allow update APPLICATION_TYPE_FIELD ?

Reply:
This is reserved to allow conversations in the conversational search to add APPLICATION_TYPE_FIELD. We can remove this field in the allowedList after GA.

Reply:
> allow conversations in the conversational search to add APPLICATION_TYPE_FIELD.

Why conversation search need to update APPLICATION_TYPE_FIELD ? I think it's just saving interactions to one conversation. Will it update the conversation application type?